extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourcesJdbcRuntimeConfig.java

```diff
@@ -2,37 +2,48 @@
 
 import java.util.Map;
 
+import io.quarkus.datasource.common.runtime.DataSourceUtil;
 import io.quarkus.runtime.annotations.ConfigDocMapKey;
 import io.quarkus.runtime.annotations.ConfigDocSection;
 import io.quarkus.runtime.annotations.ConfigGroup;
-import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefaults;
+import io.smallrye.config.WithParentName;
 
-@ConfigRoot(name = "datasource", phase = ConfigPhase.RUN_TIME)
-public class DataSourcesJdbcRuntimeConfig {
+@ConfigMapping(prefix = "quarkus.datasource")
+@ConfigRoot(phase = ConfigPhase.RUN_TIME)
+public interface DataSourcesJdbcRuntimeConfig {
 
     /**
      * The default datasource.
      */
-    @ConfigItem
-    public DataSourceJdbcRuntimeConfig jdbc;
+    DataSourceJdbcRuntimeConfig jdbc();
 
     /**
      * Additional named datasources.
      */
     @ConfigDocSection
     @ConfigDocMapKey("datasource-name")
-    @ConfigItem(name = ConfigItem.PARENT)
-    public Map<String, DataSourceJdbcOuterNamedRuntimeConfig> namedDataSources;
+    @WithParentName
+    @WithDefaults
+    Map<String, DataSourceJdbcOuterNamedRuntimeConfig> namedDataSources();
```

Comment:
Any reason not to merge these into a single `datasources()` method?

If so, can you please add a comment here so I don't break your stuff next time I end up in this part of the code? :)